docs/content/docs/guide/incoming_webhook.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Incoming Webhook
+---
+# Incoming webhook
+
+Pipelines as Code support the concept of incoming webhook URL. Which let you
+start a PipelineRun in a Repository by a URL and a shared secret rather than
+having to generate a new code iteration.
+
+## Incoming Webhook URL
+
+You need to set your incoming match your Repository CRD, in your match you
+specify a reference Secret which will be used as a shared secret and the
+branches targetted by the incoming webhook.
+
+{{< hint danger >}}
+you will need to have a git_provider spec to specify a token when using the
+github-apps method the same way we are doing for github-webhook method. Refer to
+the [github webhook documentation](/docs/install/github_webhook) for how to set
+this up.
+{{< /hint >}}
+
+Here is an example of a Repository CRD matching the target branch main:
+
+```yaml
+---
+apiVersion: "pipelinesascode.tekton.dev/v1alpha1"
+kind: Repository
+metadata:
+  name: repo
+  namespace: ns
+spec:
+  url: "https://github.com/owner/repo"
+  git_provider:
+    type: github
+    secret:
+      name: "owner-token"
+  incoming:
+    - targets:
+      - main
+      secret:
+        name: repo-incoming-secret
+      type: webhook-url
+```
+
+a secret named `repo-incoming-secret` will have this value:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: repo-incoming-secret
+  namespace: ns
+type: Opaque
+stringData:
+  secret: very-secure-shared-secret
+```
+
+after setting this up, you will be able to trigger a PipelineRun called
+`pipelienrun1` which will be located in the `.tekton` directory of the Git repo
+`https://github.com/owner/repo`. As an example here is the full curl snippet:
+
+```shell
+curl -X POST https://control.pac.url/incoming?secret=very-secure-secret&repository=repo,branch=main
```

Comment:
we are passing actual secret here 🤔 
i was thinking we would pass secret name and pac will use the token from there

Reply:
That's the webhook secret, the same way we have a webhook secret on other methods... it's actually a pattern for webhooks based URL.